site/shared.templ

```diff
@@ -101,6 +101,90 @@ templ headerTopLevelLinks(r *http.Request) {
 	@headerTopLevelLink(r, "Bundler")
 }
 
+type SiteSearchSignals struct {
+	Search string `json:"siteSearch"`
+	SearchFetching bool `json:"searchFetching"`
+	OpenSearchResults bool `json:"openSearchResults"`
+}
+
+type SearchResult struct {
+	ID    string  `json:"id"`
+	Title string `json:"title"`
+	Score float64 `json:"score"`
+	Fragment string `json:"fragment"`
+}
+
+const SearchFromHeaderKey = "X-Search-From-Header"
+
+templ SiteSearch(signals *SiteSearchSignals, searchResults []SearchResult, fromHeader bool) {
+	<div data-signals={templ.JSONString(signals)} class="w-full">
+		<div class="flex flex-nowrap w-full">
+			<input 
+				id="site_search_input" 
+				class="flex-1 input input-bordered h-8 w-full" 
+				type="search"
+				placeholder="Search..."
+				data-bind="siteSearch"
+				data-on-input__debounce.300ms={ fmt.Sprintf(
+    				"$siteSearch.length > 1 && @get('/api/search', %s)", 
+   					 fmt.Sprintf(`{"headers": {"%s": "%t"}}`, SearchFromHeaderKey, fromHeader),
+				)}
+				data-indicator="$searchFetching"
+			/>
+			@sseIndicator("searchFetching")
```

Comment:
Can you please style the indicator so this it doesn’t cause jank due to its horizontal space when visible?

![screenshot-LRmqRfTQ@2x](https://github.com/user-attachments/assets/96cfee9e-5914-4ed8-8c7a-3cb56f4b1f11)

Reply:
Fixed in: https://github.com/starfederation/datastar/pull/522/commits/ee297003c31024a05dae42f27b4e1ddd7f6a6fd8